### PR TITLE
[AIRFLOW-1729] DagBag Improve time taken in collect_dags

### PR DESCRIPTION
**Pull request by q2w**
Make sure you have checked _all_ steps below.

### JIRA
- [ ] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "[AIRFLOW-XXX] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-1729

### Description
- [ ] Here are some details about my PR, including screenshots of any UI changes:
The change will remove unnecessary accumulation of .airflowignore file entries. 

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
Trivial change

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

- [ ] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/models.py [hunk 1]
```diff
@@ -446,11 +446,11 @@ def collect_dags(
...
             self.process_file(dag_folder, only_if_updated=only_if_updated)
         elif os.path.isdir(dag_folder):
-            patterns = []
             for root, dirs, files in os.walk(dag_folder, followlinks=True):
-                ignore_file = [f for f in files if f == '.airflowignore']
-                if ignore_file:
-                    f = open(os.path.join(root, ignore_file[0]), 'r')
+                patterns = []
+                ignore_file = os.path.join(root, '.airflowignore')
+                if os.path.exists(ignore_file):
```
Comment (artwr) [inline airflow/models.py]: Can I recommend `os.path.isfile` instead? `exists` could return true for a directory.

File: airflow/models.py [hunk 2]
```diff
@@ -446,11 +446,11 @@ def collect_dags(
...
+                    f = open(ignore_file, 'r')
```
Comment (artwr) [inline airflow/models.py]: maybe we could use a `with` context manager here, it will take care of closing the file in case of an exception.

**Review by artwr — CHANGES_REQUESTED** [review]
Some minor changes.

File: airflow/models.py (hunk 1 above)
Comment (q2w) [reply airflow/models.py]: it seems good.. thanks

File: airflow/models.py (hunk 2 above)
Comment (q2w) [reply airflow/models.py]: thanks...

**Review by q2w — COMMENTED** [review]
suggested changes done. Thanks.

**Comment by artwr** [comment]
@q2w, LGTM. One last request, can you rebase and squash your commits add [AIRFLOW-1729] to your commit message?

**Comment by q2w** [comment]
@artwr  i have created an alternative clean pull request..
https://github.com/apache/incubator-airflow/pull/3171

**Comment by Fokko** [comment]
Thanks @q2w. Could you close this PR? I've merged #3171

**Pull request closed without merging**